### PR TITLE
Redirects from old guides to current ones

### LINKED ISSUE #152: Old guide redirects to the new guides
The API redirects are handled via json and code, but we need a manual list of guide redirects as well.

### PR DESCRIPTION
**Pull request by JoelMarcey**
Fixes #152

### CODE REVIEW

File: src/site/LegacyRedirectController.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    $id = $this->getRequiredStringParam('legacy_id');
+    // Since the API redirects are quite specific, see if we are redirecting
+    // from there first.
+    $url = idx(APILegacyRedirectData::getIndex(), $id);
+    if ($url) {
+      throw new RedirectException($url);
+    } else {
+      // If not, iterate through our manual redirects and see if we have
+      // $from being a string part of $id. If so, then we can redirect to the
+      // apporpriate place.
```
Comment (fredemmott) [inline src/site/LegacyRedirectController.php]: speling

File: src/site/LegacyRedirectController.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+final class LegacyRedirectController extends WebController {
+  public async function respond(): Awaitable<void> {
+    $id = $this->getRequiredStringParam('legacy_id');
+    // Since the API redirects are quite specific, see if we are redirecting
+    // from there first.
+    $url = idx(APILegacyRedirectData::getIndex(), $id);
+    if ($url) {
+      throw new RedirectException($url);
+    } else {
```
Comment (fredemmott) [inline src/site/LegacyRedirectController.php]: Remove `else {`, unindent the whole block. else is unneeded as the main block throws. Reduced indentation == easier to read

File: src/site/LegacyRedirectController.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      foreach ($this->getManualRedirectData() as $from => $to) {
+        if (strpos($id, $from) !== false) {
+          throw new RedirectException($to);
+        }
+      }
+    }
+    throw new HTTPNotFoundException();
+  }
+
+  private function getManualRedirectData(): array<string, string> {
```
Comment (fredemmott) [inline src/site/LegacyRedirectController.php]: map<string, string> ?

File: src/site/LegacyRedirectController.php (hunk 1 above)
Comment (JoelMarcey) [inline src/site/LegacyRedirectController.php]: I am going to assume you spelling spelling wrong is a troll on me :)

File: src/site/LegacyRedirectController.php (hunk 3 above)
Comment (JoelMarcey) [inline src/site/LegacyRedirectController.php]: :+1:

File: src/site/LegacyRedirectController.php (hunk 2 above)
Comment (JoelMarcey) [inline src/site/LegacyRedirectController.php]: :+1:

**Comment by JoelMarcey** [comment]
Well that didn't work

**Comment by JoelMarcey** [comment]
Ok. that looks like the feedback is in.

**Pull request merged**